packages/compass-e2e-tests/helpers/selectors.js

```diff
@@ -18,7 +18,7 @@ module.exports = {
   ShellOutput:
     '[data-test-id="shell-content"] [class^=mongosh-shell-output-line] pre',
   ShellLoader:
-    '[data-test-id="shell-content"] [class^=mongosh-shell-loader-shell-loader]',
+    '[data-test-id="shell-loader-in-progress"] [name="shell-loader-in-progress-text"]',
```

Comment:
We hide this text when shell is not collapsed, right? Is it fine that we are relying on it here instead of the previous spinner?

Reply:
I wanted to use this selector since the other one exists in the `mongosh` repo and might be a little harder to change/keep track of. Looking at it more it is using an internal state in mongosh to show this class, so it isn't quite testing the same thing now: 
https://github.com/mongodb-js/mongosh/blob/main/packages/browser-repl/src/components/shell.tsx#L346
Maybe we should go back to the old one? As you mentioned we don't show this text when the shell is collapsed so that could maybe cause an issue with a test if it's looking for this, and the command is in progress, and the shell is somehow collapsed. This selector is used in tests to check that mongosh properly runs commands. Mostly I wanted to reduce brittleness of this test from changes outside the repo.

Reply:
I am pretty confused now 😆 Unless I'm missing something I think it's the other way around: we don't show this thing when shell is expanded (and this is how all the tests are using shell, in expanded state):

https://github.com/mongodb-js/compass/blob/3e2bf2f91e364cc0f1c887e14347d414b152f9ce/packages/compass-shell/src/components/shell-header/shell-header.jsx#L45-L55

Tbh I am not sure how the tests are passing right now because I am pretty sure that [this check](https://github.com/mongodb-js/compass/pull/2438/files#diff-4bc6470c93d0244014ed60103a28d13485d6e0e7e8d5957389a071204f48e6ceR18-R26) in `shell-eval` helper should always be skipped with this change. So the only thing that allows us to wait for shell eval result is this 50ms delay that is there

Reply:
> Maybe we can just add an extra test id class that is purely connected to the in progress state.

Totally makes sense, let's do that

Reply:
I ended up reverting to the previous behavior just to minimize changes in this PR if that's cool, there was another case where we test the html internals of mongosh for things so it's better to probably address them all at once if we want to.

Reply:
Yeah, I would be up for that too! We can make browser-repl side of things more testing friendly further down the road. Thanks for looking into this!